internal/test/framework/framework.go

```diff
@@ -52,8 +82,16 @@ func initCIvars() error {
 	return nil
 }
 
-// Setup creates and initializes a variable amount of Windows VMs
-func (f *TestFramework) Setup(vmCount int) error {
+// Setup creates and initializes a variable amount of Windows VMs. If the array of credentials are passed then it will
+// be used in lieu of creating new VMs. If skipVMsetup is true then it will result in the VM setup not being run. These
+// two options are mainly used during test development.
+func (f *TestFramework) Setup(vmCount int, credentials []*types.Credentials, skipVMsetup bool) error {
+	if credentials != nil {
+		if len(credentials) != vmCount {
```

Comment:
Is this a requirement? I am fine with the assumption that we won't do individual node creation and deletion, example we create all the vms provided in vmCount or skip all of them.

Reply:
I am not sure what you are asking for here. Do you want the check removed? If the check is removed then we need to be support mixing and matching of using existing VMs with framework created ones and I did not want to go down that rabbit hole for some thing that we are using for helping with development.